Custom Properties blog (#465)
ritterim/ritterim.github.io: _posts/2021-11-24-custom-properties.md

---
layout: post

> title: "Custom Properties"

Reviewer: add CSS for better SEO - "CSS Custom Properties"

date: 2021-11-24 12:33:53
tags:
- css
- Sass
categories:
- css
twitter_text: Custom Properties allow you to change the value of a variable in your css or Sass. Find out how.
authors: "Ted Krueger"
image:
image_url:
image_credit:
---

> Have you ever wanted to change the value of a Sass variable based on screen size, a state, or for any other reason that might require changing a Sass variable? Well you can’t. Not with a Sass variable anyway, but you can by instead using a custom property. These are sometimes called css variables.

Reviewer: "but you can by instead using" is kind of awkward - what about removing "instead"?

> Have you ever wanted to change the value of a Sass variable based on screen size, a state, or for any other reason that might require changing a Sass variable? Well you can’t. Not with a Sass variable anyway, but you can by using a css custom property. These are sometimes called css variables.

Reviewer 2: ```suggestion
Have you ever wanted to change the value of a Sass variable based on screen size, a state, or for any other reason that might require changing a Sass variable? Well you can’t. Not with a Sass variable anyway. But you can by using a custom CSS property. These are sometimes called CSS variables.
```

There have been times where I have defined a variable in Sass and then wanted to change it when that element is `.active` for example. Something like:

```
.element {
  $background: red;
  background-color: $background;

  &.active {
    $background: blue;
  }
}
```

>  It was always frustrating that I couldn’t do this. Then I found out about custom properties. They work in standard css as well as Sass(scss files). It’s pretty awesome because now you can totally change the value of a variable. By using custom properties, the above code becomes:

Reviewer: space between Sass and (scss

>  It was always frustrating that I couldn’t do this. Then I found out about custom properties. They work in standard css as well as Sass (scss files). It’s pretty awesome because now you can totally change the value of a variable. By using custom properties, the above code becomes:

Reviewer 2: ```suggestion
 It was always frustrating that I couldn’t do this. Then I found out about custom properties. They work in standard CSS as well as Sass (scss files). It’s pretty awesome because now you can totally change the value of a variable. By using custom properties, the above code becomes:
```

```
.element {
  --background: red;
  background-color: var(--background);

  &.active {
    --background: blue;
  }
}
```

Notice the difference between how the two variables are used. When using a Sass variable, you define it with a `$` whereas with custom properties you need to start them with a double dash (`--`).

Sass variable: `$background: red`

> CSS Custom Property: `--background: red`

Reviewer 2: ```suggestion
CSS custom property: `--background: red`
```

> You use them in your styles differently as well. Sass variables are more straight forward, you simply add the same variable into the style value but with a custom property you need to wrap it in a `var()`. 

Reviewer 2: ```suggestion
You use them in your styles differently as well. Sass variables are more straightforward. You simply add the same variable into the style value, but with a custom property, you need to wrap it in a `var()`. 
```

> You use them in your styles differently as well. Sass variables are more straight forward, you simply add the same variable into the style value but with a custom property you need to wrap it in a `var()`. 

Reviewer 3: - Would it be good to explain `var()` a bit? something along the lines of: 
"... to set the value of a CSS property to a CSS custom property, you must return the custom property's value using the CSS `var()` function."

Author: Good catch, Ryan. Yeah, I think that would be a good change.

Author: I changed your suggestion a bit, @mrtrimble. I don't know if the custom property's value is what we would be adding the the `var` function but rather the variable name.

Reviewer 3: Gotcha, yeah I think you made it more succinct!

> Once you get the hang of it, you’ll never go back. I find that when inspecting websites lately, I notice things like colors and sizes being defined by custom properties a lot more. I think this has to do with how flexible custom properties are. I know I’ve been trying to use them more but I try to learn why something is better before I dive right in. So far, I can’t say that I have seen a situation where a Sass variable would be more beneficial than a custom property, so why not just use custom properties? 

Reviewer 2: ```suggestion
Once you get the hang of it, you’ll never go back. I find that when inspecting websites lately, I notice things like colors and sizes being defined by custom properties a lot more. I think this has to do with how flexible custom properties are. I know I’ve been trying to use them more, but I try to learn why something is better before I dive right in. So far, I can’t say that I have seen a situation where a Sass variable would be more beneficial than a custom property, so why not just use custom properties? 
```

> Want to learn more about Custom Properties? <a href="https://css-tricks.com/" target="_blank">CSS-TRICKS</a> literally wrote <a href="https://css-tricks.com/a-complete-guide-to-custom-properties/" target="_blank">A Complete Guide to Custom Properties</a>.

Reviewer 2: ```suggestion
Want to learn more about custom properties? <a href="https://css-tricks.com/" target="_blank">CSS-TRICKS</a> literally wrote <a href="https://css-tricks.com/a-complete-guide-to-custom-properties/" target="_blank">A Complete Guide to Custom Properties</a>.
```
